syntax = "proto3";
package Protocol;

import "Enum.proto";
import "Struct.proto";

option csharp_namespace = "Google.Protobuf.Protocol";

enum PacketId
{
	PKT_NONE = 0;
	PKT_C2S_ENTER_GAME = 1000;
	PKT_S2C_ENTER_GAME = 1001;
	
	PKT_S2C_SPAWN = 2000;
	PKT_S2C_DESPAWN = 2001;	
	PKT_C2S_MOVE = 2002;
	PKT_S2C_MOVE = 2003;
}

message PlayerInfo
{
	int32 playerId = 1;
	string name = 2;
	int32 posX = 3;
	int32 posY = 4;
}

message C2S_ENTER_GAME {  }
message S2C_ENTER_GAME
{
	PlayerInfo player = 1;
}
message S2C_SPAWN { }
message S2C_DESPAWN { } 
message C2S_MOVE { }
message S2C_MOVE { }
